scripts/list_init_hooks.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/python
+#
+# Scripts for print all init hooks in f9-kernel.
+# Copyright (c) 2016 The F9 Microkernel Project. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+#
+# Description:
+#       At now this script will print by hook level,
+#       but is not the real seq when kernel perform run_init_hook.
+#
+# TODO:
+#       read f9.map and print in kernel hooking sequence
+#
+
+# ----------------- LIBRARIES --------------------
+
+import os
+import re
+import sys
+import subprocess
+from collections import OrderedDict
+
+# ------------------- CONST --------------------
+
+# Define in kernel/init_hook.h
+HOOK_LEVEL = OrderedDict([])
+HOOK_LEVEL.update({'INIT_LEVEL_EARLIEST': 1})
+HOOK_LEVEL.update({'INIT_LEVEL_PLATFORM_EARLY': 0x1000})
+HOOK_LEVEL.update({'INIT_LEVEL_PLATFORM': 0x2000})
+HOOK_LEVEL.update({'INIT_LEVEL_KERNEL_EARLY': 0x3000})
+HOOK_LEVEL.update({'INIT_LEVEL_KERNEL': 0x4000})
+HOOK_LEVEL.update({'INIT_LEVEL_LAST': 0xFFFFFFFF})
+
+# Dynamiclly adding variable on the fly
+for hook in HOOK_LEVEL:
+    exec('%s = %d' % (hook, HOOK_LEVEL[hook]))
+
+
+# ----------------- CLASS --------------------
+
+class hook:
+
+    def __init__(self, s):
+        s = s.replace(' ', '')
+        self.files, hooks = s.split(':')
+
+        # Regex for splitting function and level
+        regex = r"\((.*?),(.*?)\)"
```

Comment:
This will be a big problem since initializer element should be constant, according to c99 spec,
this mean we have these choice to init element.

```
6.4.4 Constants

Syntax

constant:

integer-constant       (e.g. 4, 42L)
floating-constant      (e.g. 0.345, .7)
enumeration-constant   (stuff in enums)
character-constant     (e.g. 'c', '\0')
```
```
6.6 Constant expressions

(7) More latitude is permitted for constant expressions in initializers. Such a constant expression shall be, or evaluate to, one of the following:

— an arithmetic constant expression,

— a null pointer constant,

— an address constant, or

— an address constant for an object type plus or minus an integer constant expression.

(8) An arithmetic constant expression shall have arithmetic type and shall only have operands that are integer constants, floating constants, enumeration constants, character constants, and sizeof expressions. Cast operators in an arithmetic constant expression shall only convert arithmetic types to arithmetic types, except as part of an operand to a sizeof operator whose result is an integer constant.
```

That mean we can have these valid input, but can't valuate in python
```c
enum foobar {
    foo = 5,
    bar = 10,
};

INIT_HOOK(function, (int) &thread_init_subsys);
INIT_HOOK(function, bar);
```